tsdb/config/config.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdbconfig
+
+import (
+	"time"
+
+	"github.com/alecthomas/units"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/tsdb"
+)
+
+// Options is tsdb.Option version with defined units.
+// This is required as tsdb.Option fields are unit agnostic (time).
+type Options struct {
+	WALSegmentSize         units.Base2Bytes
+	RetentionDuration      model.Duration
+	MaxBytes               units.Base2Bytes
+	NoLockfile             bool
+	AllowOverlappingBlocks bool
+	WALCompression         bool
+	StripeSize             int
+	MinBlockDuration       model.Duration
+	MaxBlockDuration       model.Duration
+}
+
+func (opts Options) ToTSDBOptions() tsdb.Options {
+	return tsdb.Options{
+		WALSegmentSize:         int(opts.WALSegmentSize),
+		RetentionDuration:      int64(time.Duration(opts.RetentionDuration) / time.Millisecond),
```

Comment:
I'm not sure about this, this is putting units into the tsdb again.

The TSDB does not need this code. Can we not handle this in main?

Reply:
This comment hasn't been addressed.

Reply:
We cannot, `web` needs this.

Reply:
Also it's not `tsdb` just separate package within `tsdb/` dir (:

Reply:
This logic doesn't belong under tsdb one way or the other, as nothing under tsdb should care about units nor should any of tsdb depend on model.

I think any rendering-time web conversions should be done in web.

Reply:
> Also it's not tsdb just separate package within tsdb/ dir (:

That's still tsdb as far as I'm concerned.

Reply:
It's not only rendering time, it's also convenient flag parsing based on units. 

So, where would be the correct place for it? As separate `pkg/` package?

Reply:
That'd be one option. Possibly we want to split the parsing and the rendering here though.

Reply:
cool, will take a look what we can do.

Reply:
I think I found nice solution, let me know what you think. The struct with units is now fully in main.

Reply:
That looks good.